volatility3/framework/plugins/mac/list_files.py

```diff
@@ -44,28 +44,34 @@ def _vnode_name(cls, vnode: interfaces.objects.ObjectInterface) -> Optional[str]
         return v_name
 
     @classmethod
-    def _get_parent(cls, vnode):
-        parent = None
-
+    def _get_parent(cls, context, vnode):
         # root entries do not have parents
         # and parents of normal files can be smeared
         try:
-            parent = vnode.v_parent
+            parent = vnode.v_parent.dereference()
         except exceptions.InvalidAddressException:
-            pass
+            return None
+
+        if parent and not context.layers[vnode.vol.native_layer_name].is_valid(parent.vol.offset,
+                                                                               parent.vol.size):
+            return None
 
         return parent
 
     @classmethod
-    def _add_vnode(cls, vnode, loop_vnodes):
+    def _add_vnode(cls, context, vnode, loop_vnodes):
```

Comment:
This is technically an API change depending on whether `_add_vnode` is classed as a public method or a private one?  If vnode is an object, you could use vnode's context.  I don't see too much problem in doing that, because I can't consider a situation where someone would have made a vnode in one context and then want to pass in a different context?